src/mongo/db/commands/feature_compatibility_version.cpp

```diff
@@ -207,8 +207,6 @@ void FeatureCompatibilityVersion::set(OperationContext* txn, StringData version)
         // featureCompatibilityVersion to 3.4. This is a new index version that isn't supported by
         // versions of MongoDB earlier than 3.4 that will cause 3.2 secondaries to crash when it is
         // replicated.
-        std::vector<BSONObj> indexSpecs{k32IncompatibleIndexSpec};
```

Comment:
@visemet - I suspect that since this object has non-trivial constructor/destructor semantics, that the compiler must assume that simply creating the object could be meaningful even if it is not used, and therefore it does not issue a diagnostic.